### ISSUE TITLE
namespace check fails for exports with keyMirror

### ISSUE THREAD

**Issue by mjrussell** [benmosher/eslint-plugin-import#147]
Its common in react to use keyMirror instead of strings for constants. Currently both of these cause a warning with the namespace rule enabled:

```js
import keyMirror from 'keymirror';
module.exports = keyMirror({
  MY_CONSTANT: null,
});
```

Above may be due to #71 

```js
import keyMirror from 'keymirror';
export default keyMirror({
  MY_CONSTANT: null,
});
```

**Comment by mjrussell** [comment]
Updated description to reference correct issue

**Comment by benmosher** [comment]
Ah, yes. It doesn't evaluate the modules, just looks for f'real named exports. 

I also used keyMirror briefly, but have been moving toward using ES6 named exports of Symbols. 

Do you have any suggestions for how I could resolve this? I'm not sure what o could do, short of hard coding this as a special case. I'm open to suggestions, though. 

I have been considering making plugins for building the namespaces, to support Babel's CommonJS format in addition to ES6 modules. I can imagine this being a plugin as well, potentially.

**Comment by mjrussell** [comment]
Ah yes now it makes sense why this is hard after realizing you don't do any evaluation :smile:  (hence being a linter)

Yeah it seems to me the only thing you could do would be a complete hack in seeing thats using keymirror and diving into inspect the object argument. Even this would have the potential to be abused because you have no guarantees that keymirror will behave properly and return the keys how they are passed in.

Unfortunately Symbols wont work for me since they are discouraged in Redux (because they are not serializable), but I may look into a method with strings because I really like this sanity check.

**Comment by benmosher** [comment]
There is indeed always just explicitly writing
```js
export const MY_CONSTANT = "MY_CONSTANT"
           , ANOTHER_CONSTANT = "ANOTHER_CONSTANT";
// repeat ad infinitum
```

which will be properly linted. 

Once I started using Sublime and got all tokens as autocomplete tokens, this got a lot quicker, and `keyMirror` became much less enticing. 😁

**Comment by benmosher** [comment]
Closing, captured essence in #153.
